tauri-app/src-tauri/src/commands/fork.rs

```diff
@@ -0,0 +1,197 @@
+use forker::*;
+use once_cell::sync::Lazy;
+use rain_orderbook_common::error::{decode_error, DecodedErrorType};
+use revm::primitives::Bytes;
+use serde_bytes::ByteBuf;
+use std::{collections::HashMap, sync::Mutex};
+
+/// static hashmap of fork evm instances, used for caching instances between runs
+pub static FORKS: Lazy<Mutex<HashMap<String, ForkedEvm>>> =
+    Lazy::new(|| Mutex::new(HashMap::new()));
+
+#[tauri::command]
+pub async fn fork_call(
+    fork_url: &str,
+    fork_block_number: u64,
+    from_address: ByteBuf,
+    to_address: ByteBuf,
+    calldata: ByteBuf,
+) -> Result<Result<Bytes, DecodedErrorType>, String> {
+    // build key from fork url and block number
+    let key = fork_url.to_owned() + &fork_block_number.to_string();
+
+    let is_cached = { FORKS.lock().unwrap().contains_key(&key) };
+
+    let result = if is_cached {
+        let mut forks = FORKS.lock().unwrap();
+        let forked_evm = forks.get_mut(&key).unwrap();
```

Comment:
no unwrap

Reply:
this unwrap is safe since it is checked before entring this arm of if statement